addons/mail/static/src/messaging/entity/attachment/attachment.js

```diff
@@ -0,0 +1,305 @@
+                        token = token.substring(0, amp);
+                    }
+                }
+                return `https://www.youtube.com/embed/${token}`;
+            }
+            if (fileType === 'video') {
+                return `/web/image/${this.id}?model=ir.attachment`;
+            }
+            return undefined;
+        }
+
+        /**
+         * @returns {string}
+         */
+        get displayName() {
+            return this.name || this.filename;
+        }
+
+        /**
+         * @returns {string}
+         */
+        get extension() {
+            return this.filename && this.filename.split('.').pop();
+        }
+
+        /**
+         * @returns {string}
+         */
+        get fileType() {
+            if (this.type === 'url' && !this.url) {
+                return undefined;
+            } else if (!this.mimetype) {
+                return undefined;
+            }
+            const match = this.type === 'url'
+                ? this.url.match('(youtu|.png|.jpg|.gif)')
+                : this.mimetype.match('(image|video|application/pdf|text)');
+            if (!match) {
+                return undefined;
+            }
+            if (match[1].match('(.png|.jpg|.gif)')) {
+                return 'image';
+            }
+            return match[1];
+        }
+
+        /**
+         * @returns {boolean}
+         */
+        get isLinkedToComposer() {
+            return this.composers.length > 0;
+        }
+
+        /**
+         * @returns {boolean}
+         */
+        get isTextFile() {
+            const fileType = this.fileType;
+            return (fileType && fileType.indexOf('text') !== -1) || false;
+        }
+
+        /**
+         * @returns {boolean}
+         */
+        get isViewable() {
+            const mediaType = this.mediaType;
+            return (
+                mediaType === 'image' ||
+                mediaType === 'video' ||
+                this.mimetype === 'application/pdf' ||
+                this.isTextFile
+            );
+        }
+
+        /**
+         * @returns {string}
+         */
+        get mediaType() {
+            return this.mimetype && this.mimetype.split('/').shift();
+        }
+
+        /**
+         * Unlink the provided attachment.
+         */
+        async remove() {
+            await this.env.rpc({
+                model: 'ir.attachment',
+                method: 'unlink',
+                args: [this.id],
+            }, { shadow: true });
+            this.delete();
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @override
+         */
+        _createInstanceLocalId(data) {
+            const { id, isTemporary = false } = data;
+            if (isTemporary) {
+                return `${this.constructor.localId}_${nextTemporaryId}`;
```

Comment:
Shouldn't it use getAttachmentNextTemporaryId?

Reply:
no because otherwise temporary id (data) doesn't match temporary id in its local id.
For instance, we don't want `ir.attachment_-1` having a local id `ir.attachment_-2`.